core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -313,9 +265,6 @@ public SparkSqlDialect(SqlDialect.Context context) {
             timeUnitNode.getParserPosition());
         SqlFloorFunction.unparseDatetimeFunction(writer, call2, "DATE_TRUNC", false);
         break;
-      case TRIM:
```

Comment:
How we are taking care of LTRIM, RTRIM and REGEXP_REPLACE ?

Reply:
all the **TRIM**, **LTRIM**, **RTRIM** are converted to the Std operator **TRIM** in the canonical form and are therefore covered by the **TRIM** operator's unparse itself. Since QA wants TRIM to be TRIM itself and we cant identify whether if its LTRIM or TRIM(LEADING) in the dialect file it is kept as TRIM(leading) itself without converting any further.

since **REGEXP_REPLACE** in spark only takes three arguments and the **REGEXP_REPLACE** operator in SqlOperatorTable takes upto five arguments,  need to come up with a new solution to handle the additional arguments. The new solution will be raised in a new MR once the solution is finalized.